### PR TITLE
`muxink` crate

### PR DESCRIPTION
**Pull request by marc-casperlabs**
This is the `muxink` crate, the basis for backpressuring networking to be merged into the 1.6 feature.

Not much more to write in this PR, as the crate documentation should cover the details -- if it does not, please comment, so it can be fixed.

### CODE REVIEW

File: muxink/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+[package]
+name = "muxink"
+version = "0.1.0"
+edition = "2021"
+
+[dependencies]
+anyhow = "1.0.57"
```
Comment (goral09) [inline muxink/Cargo.toml]: Can we get rid of this dependency? It always seemed to me that it's a good one if you write a CLI tools and don't bother about handling errors properly. `muxink` is a library and probably a very-well written so it does (🤞 ) return all results in a typed fashion?

Comment (marc-casperlabs) [reply muxink/Cargo.toml]: Good catch. It's actually not in use anymore, it was used in earlier versions.

Comment (marc-casperlabs) [reply muxink/Cargo.toml]: Fixed in 2c98125a0

File: muxink/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+version = "0.1.0"
+edition = "2021"
+
+[dependencies]
+bincode = { version = "1.3.3", optional = true }
+bytes = "1.1.0"
+futures = "0.3.21"
+serde = { version = "1.0.138", optional = true }
+thiserror = "1.0.31"
+tokio = { version = "1", features = [ "full" ] }      # TODO: Reduce features.
```
Comment (goral09) [inline muxink/Cargo.toml]: While we're here - do you plan to reduce these features as part of the PR?

Comment (marc-casperlabs) [reply muxink/Cargo.toml]: Good point, fixed in 8c6374d77.

**Review by dwerner — APPROVED** [review]

**Pull request merged**